### DISCUSSION TITLE
Can't change sorting or exclude app?

### DISCUSSION THREAD

**Discussion by mstieranka** [johnlindquist/kit#1364] [Q&A]
I'm setting up Script Kit and I'm frustrated by the fact that the in the App Launcher, when inputting `code` it's suggesting "Coverage Details" over "Visual Studio Code". (Even though the former doesn't even open! I get the point of being able to type the first letters, but I feel like whole words should be prioritized? or even some sort based on usage?)

From what I understand, I can't change the sorting of the arg function (yet?), but I also tried adding an exclusion list to exclude this specific app, but that also does nothing. Am I doing something wrong? I feel like I tried everything.

Modified App Launcher here:

```typescript
// Shortcut: command Space
import "@johnlindquist/kit"
    
// Menu: App Launcher
// Description: Search for an app then launch it
// Author: John Lindquist
// Twitter: @johnlindquist

const exclusionList = [
  "/System/Library/CoreServices/Coverage Details.app",
];

let createChoices = async () => {
  let apps = await fileSearch("", {
    onlyin: "/",
    kind: "application",
  })

  let prefs = await fileSearch("", {
    onlyin: "/",
    kind: "preferences",
  })

  let group = (path: RegExp) => (apps: string[]) =>
    apps
      .filter(app => app.match(path))
      .filter(app => !exclusionList.includes(app))
      .sort((a, b) => {
        let aName = a.replace(/.*\//, "")
        let bName = b.replace(/.*\//, "")

        return aName > bName ? 1 : aName < bName ? -1 : 0
      })

  return [
    ...group(/^\/Applications\/(?!Utilities)/)(apps),
    ...group(/\.prefPane$/)(prefs),
    ...group(/^\/Applications\/Utilities/)(apps),
    ...group(/System/)(apps),
    ...group(/Users/)(apps),
  ].map(value => {
    return {
      name: value.split("/").pop().replace(".app", ""),
      value,
      description: value,
    }
  })
}

let appsDb = await db("apps", async () => ({
  choices: await createChoices(),
}))

let app = await arg("Select app:", appsDb.choices)
let command = `open -a "${app}"`
if (app.endsWith(".prefPane")) {
  command = `open ${app}`
}
exec(command)
```

**Comment by mstieranka** [comment]
Seems like I fixed it by manually deleting the database file and letting it recreate (in `.kenv/db`), but the fact that this isn't documented anywhere (the `db` function as a whole is not in the API reference) seems like a blind spot.

**Reply by johnlindquist** [reply]
Thanks for bringing this up! I only have "exact match to top" set for "startsWith(input)".
I'll add a `includes(` ${input}`)` so it brings up exact matches following a space.
